### ISSUE TITLE
How do I turn off honeycomb during the circleci run of my django app?

### ISSUE THREAD

**Issue by simkimsia** [honeycombio/beeline-python#125]
So far, I have been able to turn off honeycomb in my django in circleci using this way

```python
import beeline
from django.apps import AppConfig
from django.conf import settings


class CoreConfig(AppConfig):
    name = "core"

    def ready(self):
        if settings.HONEYCOMB_ON:
            beeline.init(
                writekey=settings.HONEYCOMB_API_KEY,
                dataset=settings.HONEYCOMB_DATASET,
                service_name=settings.HONEYCOMB_SERVICE_NAME,
                debug=settings.DEBUG,
            )
```

So in circleci, settings.HONEYCOMB_ON is false and it works until 2.12

but with the new 2.13 version this no longer works for me.

in my circleci, I get the following

```bash
Traceback (most recent call last):
  File "/home/circleci/repo/config/tests/test_redirect.py", line 8, in test_redirect
    response = self.client.get("/")
  File "/home/circleci/repo/venv/lib/python3.7/site-packages/django/test/client.py", line 535, in get
    response = super().get(path, data=data, secure=secure, **extra)
  File "/home/circleci/repo/venv/lib/python3.7/site-packages/django/test/client.py", line 347, in get
    **extra,
  File "/home/circleci/repo/venv/lib/python3.7/site-packages/django/test/client.py", line 422, in generic
    return self.request(**r)
  File "/home/circleci/repo/venv/lib/python3.7/site-packages/django/test/client.py", line 503, in request
    raise exc_value
  File "/home/circleci/repo/venv/lib/python3.7/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/home/circleci/repo/venv/lib/python3.7/site-packages/beeline/middleware/django/__init__.py", line 144, in __call__
    response = self.create_http_event(request)
  File "/home/circleci/repo/venv/lib/python3.7/site-packages/beeline/middleware/django/__init__.py", line 104, in create_http_event
    dr = DjangoRequest(request)
  File "/home/circleci/repo/venv/lib/python3.7/site-packages/beeline/middleware/django/__init__.py", line 12, in __init__
    beeline.get_beeline().log(request.META)
AttributeError: 'NoneType' object has no attribute 'log'

----------------------------------------------------------------------
```
 how do i turn off honeycomb during the circleci run of the my django app?

**Comment by MikeGoldsmith** [comment]
Hi @simkimsia - thanks for raising the issue. I think this is a bug and we'll try to get it resolved quickly.

If it is causing problems with your CircleCI runs, we'd recommend you downgrade to 2.12.0 for now.

**Comment by MikeGoldsmith** [comment]
@simkimsia on review, I'm not sure where `HONEYCOMB_ON` is coming from. It's not part of our libhoney / beeline SDKs. Is that a setting you're setting in your django config?

**Comment by MikeGoldsmith** [comment]
Never mind - I see it is coming from your config. However, the problem is that you're seeing a runtime error because the beeline is not initialised during testing.

We'll look into what's going on.

**Comment by simkimsia** [comment]
Or you can suggest an alternative for me to turn off honeycomb during circleci testing

That will help as well

**Comment by MikeGoldsmith** [comment]
The issue is that the django middleware does not know if the beeline has been initialised and should be checking it has been before trying to use it. There are some missing checks that can be added to bypass the beeline if it's not been initialised.

**Comment by simkimsia** [comment]
So I Guess there's nothing I can do in my end?

I'll wait till #126 is merged to production branch

**Comment by MikeGoldsmith** [comment]
The way we deal with it in our own tests is to import mock, beeline.init and then set beeline.tracer_impl with a `Mock()` - [example](https://github.com/honeycombio/beeline-python/blob/main/beeline/test_beeline.py#L15).

You can try to see if that works for you too.

**Comment by simkimsia** [comment]
Maybe I misunderstand your test case example

But my intention isn't to run a mock version of beeline in my circleci tests

I just want to avoid turning beeline on. I am okay to have it on in production hence I use env variables to turn it on via my settings

**Comment by MikeGoldsmith** [comment]
You have it correct. Unfortunately, there is not a global switch to disable honeycomb so you need to either provide a mock `tracer_impl` that effectively noop's, or with the above fix, the django middleware won't try to use an non-initialised beeline.

**Comment by simkimsia** [comment]
so if i take your advice, the code becomes


```python
import beeline
from django.apps import AppConfig
from django.conf import settings
from mock import Mock


class CoreConfig(AppConfig):
    name = "core"

    def ready(self):
        beeline.init(
            writekey=settings.HONEYCOMB_API_KEY,
            dataset=settings.HONEYCOMB_DATASET,
            service_name=settings.HONEYCOMB_SERVICE_NAME,
            debug=settings.DEBUG,
        )
        if settings.HONEYCOMB_ON is False:
            beeline.tracer_impl = Mock()

```

yes?

but beeline will still run in circleci yes? of course using a mock. Will that slow my circleci tests?

**Comment by simkimsia** [comment]
oh i have tried running locally with the mock, i think i prefer not to even turn it on when running tests.

**Comment by MikeGoldsmith** [comment]
Yeah, your above code sample is what I was thinking by using a mock. I agree it's not ideal.

#126 will address the issue so the django middleware won't attempt to use a non-initialised beeline instance.

**Comment by MikeGoldsmith** [comment]
Hey @simkimsia - we've just released v2.13.1 that includes the fix for the django middleware: https://pypi.org/project/honeycomb-beeline/

Please try it and let us know if there are any issues 👍

**Comment by simkimsia** [comment]
yes solved. thank you 🙏
